earth_enterprise/src/fusion/gecombineterrain/gecombineterrain.cpp

```diff
@@ -232,28 +234,47 @@ int main(int argc, char **argv) {
                 &khGetopt::RangeValidator<uint32, 0, 1024>);
     options.opt("read_cache_block_size", read_cache_block_size,
                 &khGetopt::RangeValidator<uint32, 1, 1024>);
-
     if (!options.processAll(argc, argv, argn)) {
       usage(progname);
     }
     if (help) {
       usage(progname);
     }
+
     if (argn == argc) {
-      usage(progname, "No input indexes specified");
+      usage(progname, "No input indices specified");
     }
+    uint cmdDefaultCPUs = CommandlineNumCPUsDefault();
+    uint numavailable = GetMaxNumJobs();
+
+    /* 
+        Choosing number of CPUs:
+
+        numcpus: initially, the command line parameter --numcpus
+	
+        CommandLineNumCPUsDefault(): defined in /etc/opt/google/systemrc via maxjobs tag
+
+        GetMaxNumJobs(): the number of available jobs, defined as the minimum value of either:
+         1. the currently available jobs or the default number of jobs i.e. kMaxNumJobs (is 8) multiplied by KH_MAX_NUM_JOBS_COEFF, which is a compiler option
+	 2. kMaxNumJobsDefault = 8
+         3. environment variable, KH_GOOGLE_MAX_NUM_JOBS, if defined
+         4. the number of currently available cpus, obtained through sysconf
+
+         The methodology	first checks if the --numcpus flag was present, if
```

Comment:
Minor: there seems to be an interesting amount of space between words on this line.